WIP:  1.4.0 packaging and versioning changes

jhsimpson opened the pull request:
This Pull Request includes packaging changes, and changes to the sql scripts, required in advance of a 1.4.0 release.

This has been testing on a new install only.  Additional testing with upgrades is ongoing.

jhsimpson committed:
Create new mysql base file

refs #8180

Update the base file used to populate the database on new installs.
Create a 1.4.0 upgrade script.
Remove the mysql_dev_*sql files that got concatenated into that upgrade script.
Alter mysql_dev.sh so it is ready for the next round of development.

jhsimpson committed:
Update version number

refs #8180

in Archivematica Common package, update the version.py for
the upcoming release.

jhsimpson committed:
updating debian packaging files for 1.4.0 release

refs #8180

updating fits dependency number
updating dependencies for dashboard
updating archivematica common dependencies
updating changelogs for last qa release

Hwesta commented on src/MCPClient/debian/copyright:
```diff
@@ -1,6 +1,6 @@
 This work was packaged for Ubuntu by:
 
-    Justin Simpson <jsimpson@artefactual.com>
+    Artefactual Systems Inc. <sysadmin@artefactual.com>
```
Do we want to use ops@artefactual.com (as in the control file) or sysadmin@artefactual.com (as here).  Are they supposed to be the same?

jhsimpson replied:
It should be sysadmin@ in all places, if there are individual emails left or ops@ anywhere, it is because I missed it, I will update the PR.

Hwesta commented on src/MCPServer/debian/postinst:
```diff
@@ -13,9 +21,14 @@ else
 fi
 
-chown -R archivematica:archivematica /var/archivematica/
+chown -R archivematica:archivematica /var/archivematica/sharedDirectory
```
Add the trailing `/` - if `/var/archivematica/sharedDirectory` is a symlink this will only change permissions on the symlink not the directory.

Hwesta commented on src/MCPServer/share/postinstSharedWithDev:
```diff
@@ -7,17 +7,10 @@ else
-sudo chown -R archivematica:archivematica $logdir
-sudo chmod -R g+s $logdir
+sudo chmod -R g+s "/var/archivematica/sharedDirectory"
+sudo chmod -R 664 "/var/archivematica/sharedDirectory"
```
These should also have a trailing / if we want to affect the directory not the (possible) symlink. Or do we want to affect the symlink, given the line below?

Hwesta commented on src/MCPrpcCLI/debian/control:
```diff
@@ -2,12 +2,14 @@ Source: archivematica-mcp-rpccli
+Build-Depends:
+  debhelper (>= 9)
+Standards-Version: 3.9.5
+X-Python-Version: 2.7
```
Glad to see the python version being specified now.

Hwesta commented on src/dashboard/debian/changelog:
```diff
@@ -1,3 +1,10 @@
+  * commit: 828eb90be1a4bfe860b67a4398e8018456695dce
+  * 1.4 qa rc3 build
+
+ -- Justin Simpson <jsimpson@artefactual.com>  Tue, 12 May 2015 17:43:42 -0800
```
Should this be Artefactual instead of Justin?

jhsimpson replied:
in the changelogs,I think it might have to be the person who signed the package.  It is my gpg key that signed the files being uploaded to launchpad, that is based on my email address.

For copyright and control files, it should be Artefactual.

We could create an Artefactual gpg key but I am not sure if that is a good idea or not.  So far each individual uses their own gpg key.

Hwesta replied:
Ah, makes sense. Easiest to leave it as individual keys for now.

jhsimpson commented:
edits made and commits in the branch rebased.

Hwesta committed:
FPR: Update PRONOM for v82

updating mysql script with new pronom id's

refs #8252

Hwesta committed:
FPR: Update PRONOM for v82

updating mysql script with new pronom id's

refs #8252

Hwesta committed:
Setup: Remove empty script

Hwesta committed:
Update gitignore

Hwesta committed:
MCPServer: Configure logging

Hwesta committed:
MCPServer: Add logging

jhsimpson committed:
removing 8252 sql script

refs #8252

update script now incorporated in main mysql file and 1.4.0 update script

jhsimpson commented:
I just merged the 8252 branch into qa/1.4.0, and made some changes to it.  Specfically, I updated the mysql and 1.4.0 files in MCPServer/share, with the new sql from 8252, and then updated mysql_dev.sh and removed the 8252 update script.

Not sure this is the cleanest way to get this change into the commit histor.

jhsimpson committed:
fixing typos in postinstall scripts

logging permisssions were not getting set correctly

Hwesta committed:
MCPServer: remove logging of print statements

mistydemeo commented:
:+1:

hakamine committed:
Fix SQL to update MCP database from 1.3 to 1.4

Hwesta committed:
Convert datetimes to timezone-aware UTC

refs #8541

Django timezone set to UTC, and adding pytz for better features. getUTCDate
returns a timezone-aware UTC datetime object, instead of a string for easier
handling.

Hwesta committed:
Remove excessive imports

Hwesta committed:
Template: Display times in localtime with JS

refs #8541

Add JS to parse a datetime and display it in local time for task times. Fix
confusing SST/EET/etc to be plain English.

mistydemeo commented:
The release has been made, do we want to merge this into qa/1.x?

Hwesta commented:
These changes are now in the packaging branch for 1.5 #380
